src/main/java/net/tiagofar78/prisonescape/game/prisonbuilding/doors/GrayDoor.java

```diff
@@ -0,0 +1,25 @@
+package net.tiagofar78.prisonescape.game.prisonbuilding.doors;
+
+import net.tiagofar78.prisonescape.game.PrisonEscapeGame;
+import net.tiagofar78.prisonescape.game.PrisonEscapePlayer;
+import net.tiagofar78.prisonescape.items.GrayKeyItem;
+import net.tiagofar78.prisonescape.items.Item;
+import net.tiagofar78.prisonescape.managers.GameManager;
+
+public class GrayDoor extends Door {
+
+    public ClickDoorReturnAction click(PrisonEscapePlayer player, Item itemHeld) {
+        PrisonEscapeGame game = GameManager.getGame();
+        boolean isOpened = isOpened();
+
+        if (game.isPolice(player))
+            return isOpened ? ClickDoorReturnAction.CLOSE_DOOR : ClickDoorReturnAction.NOTHING;
+
+        if (game.isPrisioner(player))
+            return !isOpened && itemHeld instanceof GrayKeyItem
```

Comment:
Why are you returning a ClickDoorReturnAction instead of doing the action in this method?

Reply:
Just to keep the code abstraction. This allows the calling code to decide what to do with the returned action, for example, sending a message to the player or performing any additional check.